onnxruntime/core/framework/session_state.cc

```diff
@@ -1015,15 +1015,24 @@ Status SessionState::LoadFromOrtFormat(const fbs::SessionState& fbs_session_stat
   }
 #endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD)
 
-  // lookup the hashes for any nodes we compiled. the nodes indexes for compiled nodes are not in node_indices
+  // lookup the hashes for any nodes we compiled or added during graph partitioning.
+  // These node indexes for compiled nodes as well as newly added nodes are not in node_indices
   // as they were created at runtime.
-  if (!compiled_kernel_hashes.empty()) {
-    for (const auto& node : graph_.Nodes()) {
-      if (kernel_create_info_map_.count(node.Index()) == 0) {
+  auto new_node_hashes = kernel_registry_manager.GetStaticKernelHashMap();
```

Comment:
Do we need to lookup these hashes in InferenceSession::AssignNodesToEpsFromHashesImpl as well as here? If so can we have a helper method for the lookup so we're not paying the binary size cost twice? 

I'm fine only doing it here though. Either way we will detect a node with a missing hash and fail.

Reply:
InferenceSession::AssignNodesToEpsFromHashesImpl assigns execution providers to all nodes using the hashes where as here we add all the kernels to kernel_create_info.... so yes we need this in both the places... 

this is a simple map lookup which is being done in 2 places only so did not go for a helper... I dont have a strong opinion for this, I can add a helper here